### PR TITLE
Implement Neural Model Architecture (ParserMLP)

### LINKED ISSUE #8: Implement Neural Model Architecture (ParserMLP)
### Description
Define the MLP-based dependency parser (embeddings + dense layers + softmax).

### Tasks
- [ ] Create `__init__` storing hyperparameters
- [ ] Define Keras Input layer
- [ ] Add Embedding layers (FORM + UPOS)
- [ ] Concatenate embeddings
- [ ] Add Dense hidden layers (ReLU)
- [ ] Add softmax output(s) for transitions and deprels
- [ ] Compile the model and store in `self.model`

### PR DESCRIPTION
**Pull request by byluca**
I have fully implemented the MLP-based parser in model.py.

Key Changes:

    1) Model Architecture: Implemented the Keras model with Word/POS embeddings, two hidden layers, and dual outputs       (Action and Deprel).

     2)Methods: Completed __init__, train, evaluate, and the run method for inference.

     3)Logic: The run method now correctly applies the Arc-Eager algorithm for parsing.

    4) Usage: Updated main.py with an example to train and test the model.

closes #8

### CODE REVIEW

**Review by Emilio25232 — APPROVED** [review]
Excelent!!
